File: EasyPlan.Web/App/mappers/boardMapper.js
define(['models/board', 'mappers/itemMapper', 'mappers/criterionMapper'],
    function (Board, itemMapper, criterionMapper) {

    return {
        map,
        mapToShortInfo
    }

    function map(src) {
        return new Board({
            id: src.id,
            title: src.title,
            criterions: _.map(src.criterions, criterionMapper.map),
            items: _.map(src.items, itemMapper.map)
        });
    }

    function mapToShortInfo(src) {
        return new Board({
            id: src.id,
            title: src.title,
        });
    }
});

File: EasyPlan.Web/Components/Mapper/BoardMapper.cs
using System.Linq;
using EasyPlan.DomainModel.Entities;

namespace EasyPlan.Web.Components.Mapper
{
    public static class BoardMapper
    {
        public static object Map(Board entity)
        {
            return new
            {
                id = entity.Id,
                title = entity.Title,
                items = entity.Items.Select(e => ItemMapper.Map(e)),
                criterions = entity.Criterions.Select(e => CriterionMapper.Map(e)),
            };
        }

        public static object MapToShortInfo(Board entity)
        {
            return new
            {
                id = entity.Id,
                title = entity.Title,
                createdOn = entity.CreatedOn.ToString(),
                createdBy = entity.CreatedBy,
            };
        }

        public static object MapToUsersInfo(Board entity)
        {
            return new
            {
                createdOn = entity.CreatedOn.ToString(),
                createdBy = entity.CreatedBy,
                usersInRoles = entity.Rights.Select(e => RoleMapper.Map(e))
            };
        }
    }
}